api/all/src/main/java/io/opentelemetry/api/OpenTelemetryTesting.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api;
+
+/** Utilities for use from tests related to OpenTelemetry. */
+public final class OpenTelemetryTesting {
```

Comment:
Can this live in the testing artifact, or do we need it in the API? Look like it's needed for testing the SDK itself. I wonder if the SDK could provide its own little bridge for testing, outside of the official testing artifact.

Reply:
I would rather make the `resetForTest` public than adding a new class, what do you think?

Reply:
If we can somehow document that people calling it in non-test code will not be supported, and that it isn't a part of the official public API, then I would be fine with it. I fear that if we keep it public on the API that it will get abused and we'll be stuck with it forever.

Reply:
I think it's a bit too indirect to add this only to the testing artifact, and would be impossible for custom implementations to test with then.

This is a public API, I don't think we can run away from that at least in terms of an API break perspective. I feel the `This is only meant to be used from tests which need to reconfigure {@link OpenTelemetry}.` caveat and the method name itself having ForTest is reassuring enough to me. Should I move it back to OpenTelemetry?

Reply:
If we need it to be a public API, then yes, it should be on the GlobalOpenTelemetry instance. I'm still not convinced that we need this public outside of the testing artifact, though. The testing artifact is a testing API, and if it exposed a public method that called a non-public method on the GOT, I'm not sure how that would inhibit 3rd parties from doing what they needed to do. 

I feel like I must be missing something, though. Why is it problematic to have the public method only in the testing artifact?

Reply:
One issue is the testing artifact is for the SDK and depends on it, but this method is for the API, so a bit out of scope. It could still be ok though.

But to expose a method in the testing artifact calling a package private method, I guess it'd need to be in the same package. Don't believe we could have two artifacts exposing classes from the same package for java module reasons. Presumably module system also thinks `reset` should be next to `set`, which is its sibling method :P

Reply:
The Java module thing is a good callout that I hadn't considered. Ugh. Perhaps we have to just live with it being a part of the public API, and hope people don't use it to do evil things that we don't want them to.

Reply:
Thanks - moved it back, hoping the `ForTest` is clear enough.

I did come up with a "friend class" implementation - this method could be left public but throw an exception if `new Throwable().getStackTrace()` doesn't contain a specially recognized class from the testing artifact. Think it's too annoying though.